mlx/transforms.h

```diff
@@ -6,6 +6,9 @@
 
 namespace mlx::core {
 
+/** Return whether we are during tracing for a function transformation. */
+bool currently_tracing();
+
```

Comment:
We should probably put this in `mlx::core::detail` for now just in case.. I don't see it used anywhere outside. Could even just declare+define it in `array.cpp` until we feel like we need to expose it.

Reply:
Another alternative is to move all the tracing struct to `transforms_impl.h` and then you can include that in `array.cpp` and define the `currently_tracing` there so it's not accessible.

Reply:
I went with option 2 from this comment. My initial thought was that we would actually expose the `in_tracing()` method (even in python) as it may be useful for things. But we can always do that later.